pkg/kwokctl/runtime/compose/cluster.go

```diff
@@ -474,34 +476,146 @@ func (c *Cluster) Down(ctx context.Context) error {
 	return nil
 }
 
-func (c *Cluster) Start(ctx context.Context, name string) error {
-	config, err := c.Config(ctx)
+func (c *Cluster) Start(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+	conf, err := c.Config(ctx)
 	if err != nil {
 		return err
 	}
-	conf := &config.Options
 
-	err = exec.Exec(ctx, c.Workdir(), exec.IOStreams{}, conf.Runtime, "start", c.Name()+"-"+name)
+	// TODO: nerdctl does not support 'compose start' in v1.1.0 or earlier
+	// Support in https://github.com/containerd/nerdctl/pull/1656 merge into the main branch, but there is no release
+	subcommand := "start"
+	if conf.Options.Runtime == "nerdctl" {
+		subcommand = "up"
+	}
+
+	commands, err := c.buildComposeCommands(ctx, []string{subcommand}...)
 	if err != nil {
 		return err
 	}
+
+	err = exec.Exec(ctx, c.Workdir(), exec.IOStreams{
+		ErrOut: os.Stderr,
+		Out:    os.Stderr,
+	}, commands[0], commands[1:]...)
+	if err != nil {
+		logger.Error("Failed to start cluster", err)
+	}
+
 	return nil
 }
 
-func (c *Cluster) Stop(ctx context.Context, name string) error {
-	config, err := c.Config(ctx)
+func (c *Cluster) Stop(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+	conf, err := c.Config(ctx)
+	if err != nil {
+		return err
+	}
+
+	// TODO: nerdctl does not support 'compose stop' in v1.0.0 or earlier
+	// Support in https://github.com/containerd/nerdctl/pull/1656 merge into the main branch, but there is no release
+	subcommand := "stop"
+	if conf.Options.Runtime == "nerdctl" {
+		subcommand = "down"
+	}
+
+	commands, err := c.buildComposeCommands(ctx, []string{subcommand}...)
 	if err != nil {
 		return err
 	}
-	conf := &config.Options
 
-	err = exec.Exec(ctx, c.Workdir(), exec.IOStreams{}, conf.Runtime, "stop", c.Name()+"-"+name)
+	err = exec.Exec(ctx, c.Workdir(), exec.IOStreams{
+		ErrOut: os.Stderr,
+		Out:    os.Stderr,
+	}, commands[0], commands[1:]...)
+	if err != nil {
+		logger.Error("Failed to stop cluster", err)
+	}
+	return nil
+}
+
+func (c *Cluster) startComponent(ctx context.Context, component internalversion.Component) error {
+	conf, err := c.Config(ctx)
+	if err != nil {
+		return err
+	}
+	err = exec.Exec(ctx, c.Workdir(), exec.IOStreams{}, conf.Options.Runtime, "start", c.Name()+"-"+component.Name)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func (c *Cluster) stopComponent(ctx context.Context, component internalversion.Component) error {
+	conf, err := c.Config(ctx)
+	if err != nil {
+		return err
+	}
+	err = exec.Exec(ctx, c.Workdir(), exec.IOStreams{}, conf.Options.Runtime, "stop", c.Name()+"-"+component.Name)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Cluster) StartComponents(ctx context.Context, names ...string) error {
+	cs := []internalversion.Component{}
+	for _, name := range names {
+		component, err := c.GetComponent(ctx, name)
+		if err != nil {
+			return err
+		}
+		cs = append(cs, component)
+	}
+
+	groups, err := components.GroupByLinks(cs)
```

Comment:
Here if I start only one component, such as apiserver, the calculation result may be reported as error dependent etcd does not exist, so I should pass all the components and then filter the calculation result.

Reply:
This does exist, and I understand that no matter what component the user specifies to start, the relationship depends on the state of the component, so I understand that it involves recording the actual running state of the component. Whether we only support individual component start and stop at this stage, the specific order is up to the user to control